Update tests to represent changes to cmip5-al33

charles-turner-1 opened the pull request:
- Fix to `.conda/meta.yaml` to ensure `conda install` requirements match `pip install requirements`.
- Update tests to match changes to NCI `cmip5-al33` datastore made 09/02/25.
- e2e tests all passing, save for `tests/e2e/test_datasets_representative.py::test_alignment[CordexTranslator]`.
    - I've checked all schema and I __cannot for the life of me__ find which file the translator is reading with the wrong metadata schema. I assume it's related to #284, but any and all attempts to track it down have given me metadata schema where the erroneous `day` was replaced with the correct `1day` field.
```bash
 $ pytest tests/e2e/test_datasets_representative.py::test_alignment[CordexTranslator] - AssertionError: Failed to build CordexTranslator with exception
  01 | frequency : 'day' is not valid under any of the given schemas
assert False
```

@marc-white I know you tracked down a few of these mistakes - do you have any ideas why this test might be mysteriously failing?

Other than that, we should be able to deploy this as a 1.0.1 release to fix the pin on intake 0.7

charles-turner-1 commented:
N.B: The intention is not to merge this into main, but to trigger a v1.0.1 deployment from this branch.

marc-white commented:
>       * I've checked all schema and I **cannot for the life of me** find which file the translator is reading with the wrong metadata schema. I assume it's related to [Metadata fixes for ig45 cordex #284](https://github.com/ACCESS-NRI/access-nri-intake-catalog/pull/284), but any and all attempts to track it down have given me metadata schema where the erroneous `day` was replaced with the correct `1day` field.
> 
> 
> ```shell
>  $ pytest tests/e2e/test_datasets_representative.py::test_alignment[CordexTranslator] - AssertionError: Failed to build CordexTranslator with exception
>   01 | frequency : 'day' is not valid under any of the given schemas
> assert False
> ```
> 
> @marc-white I know you tracked down a few of these mistakes - do you have any ideas why this test might be mysteriously failing?
> 

That's super weird. I just ran `metadata-validate` over all the YAML we have stored on `xp65`, and they all passed. That `01 | ` syntax is what `metadata-validate` spits out, so there is *something* failing metadata validation in that test... we just need to work out what.

marc-white commented:
I wonder if we need to alter how the exceptions are caught on L49-50 of `test_datasets_representative` to be able to pull out the full trace?

marc-white commented:
I've also noticed this branch is many commits behind main - are you happy for me to bring it up-to-date then retry? Among other things, the current branch is using a different environment for `test_end_to_end` than `main` is.

marc-white commented:
I've merged in the outstanding changes from `main`, and CI has passed. I'm running the `e2e` tests now...

I note one of the commits that was missing is labelled `199 cordex xfails (#304 )`, so I wonder if that's what was missing.

charles-turner-1 commented:
> I've also noticed this branch is many commits behind main - are you happy for me to bring it up-to-date then retry? Among other things, the current branch is using a different environment for `test_end_to_end` than `main` is.

Sorry, was screwing around with my router at home & missed this - the intention was to create a 1.0.1 release with only the meta.yaml changed from the project state at the 1.0.0 tag so that there was a 1.0.x release with the correct dependencies on available through conda.

TBH, I think it's unimportant - if the E2E tests pass, I think we can merge this & then just bump the dependencies as part of 1.1.0

charles-turner-1 commented:
> I've merged in the outstanding changes from `main`, and CI has passed. I'm running the `e2e` tests now...
> 
> I note one of the commits that was missing is labelled `199 cordex xfails (#304 )`, so I wonder if that's what was missing.

Just checked - that's the failing test.

marc-white commented:
So the end-to-end test had 48 failures :\ Where should we go from here? Given your aim to effectively do a 1.0.0 release, but with updated `cmip-al33`, do we need to do:

- Revert the merge from `main`;
- Cherry-pick out the `cordex` test fix;
- Go again with E2E tests?

charles-turner-1 commented:
Crikey... Can you send me the PBS logs? Think I had 48 come out of the cmip5 changes

marc-white commented:
@charles-turner-1 attached. 
[test_end_to_end.sh.o135126749.txt](https://github.com/user-attachments/files/18759849/test_end_to_end.sh.o135126749.txt)

charles-turner-1 commented:
Okay. 
- Most of these are related to the yaml safeloader - 38 by my count. [This PR](https://github.com/ACCESS-NRI/intake-dataframe-catalog/pull/79) should fix that. It appears somewhere, when consolidating on paths objects, we passed a path that used to be a string into a yaml.safe_read. The linked PR adds a constructor for that to the intake-dataframe-catalog - yaml.safe_read wont deserialize python objects unless they're deliberately specified to avoid executing arbitrary code when loading yaml files.( I didn't know about this restriction/capability, but it seems potentially very useful)
- The other ten failures are (I think), related to this:
```diff
-pytest -s --e2e /g/data/xp65/admin/access-nri-intake-catalog/tests/e2e
+pytest -s --e2e /home/189/ct1163/access-nri-intake-catalog/tests/e2e
```
TLDR; The tests were running against main, not this branch - hence the CMIP5 failures.

marc-white commented:
Hmmm thanks for catching the main issue, I'll re-check that. I swear I'd done a local editable install of the branch I was wanting...

charles-turner-1 commented:
Okay - quick sitrep.

I've updated my local `bin/test_end_to_end.sh` to the following:
```bash
$ cat bin/test_end_to_end.sh
#!/bin/bash -l

#PBS -P iq82
#PBS -l storage=gdata/xp65+gdata/ik11+gdata/cj50+gdata/p73+gdata/dk92+gdata/al33+gdata/rr3+gdata/fs38+gdata/oi10+gdata/hq89+gdata/py18+gdata/ig45+gdata/zz63+gdata/rt52
#PBS -q normal
#PBS -W block=true
#PBS -l walltime=00:30:00
#PBS -l mem=64gb
#PBS -l ncpus=12
#PBS -l wd
#PBS -j oe
#PBS -W umask=0022

###########################################################################################
# Copyright 2022 ACCESS-NRI and contributors. See the top-level COPYRIGHT file for details.
# SPDX-License-Identifier: Apache-2.0

# Description:
#   Generate access-nri intake metacatalog from config files

###########################################################################################

set -e

module use /g/data/xp65/public/modules
module load conda/analysis3-24.12
module load openmpi

source /home/189/ct1163/access-nri-intake-catalog/bin/venv/bin/activate

python -c "import intake_dataframe_catalog; print(intake_dataframe_catalog.__file__)"

python -m pytest -s --e2e /home/189/ct1163/access-nri-intake-catalog/tests/e2e
```

My venv site packages:
```bash 
$ pip list
Package                   Version                 Editable project location
------------------------- ----------------------- ------------------------------------------
access_nri_intake         1.0.0+21.g7743846.dirty /home/189/ct1163/access-nri-intake-catalog
annotated-types           0.7.0
attrs                     25.1.0
bokeh                     3.6.3
certifi                   2025.1.31
cf_xarray                 0.10.0
cftime                    1.6.4.post1
charset-normalizer        3.4.1
click                     8.1.8
cloudpickle               3.1.1
colorama                  0.4.6
contourpy                 1.3.1
crc32c                    2.7.1
dask                      2025.1.0
Deprecated                1.2.18
distributed               2025.1.0
donfig                    0.8.1.post1
ecgtools                  2024.7.31
fastprogress              1.0.3
fsspec                    2025.2.0
idna                      3.10
importlib_metadata        8.6.1
iniconfig                 2.0.0
intake                    2.0.8
intake_dataframe_catalog  0.3+9.ga3118be
intake-esm                2025.2.3
Jinja2                    3.1.5
joblib                    1.4.2
jsonschema                4.23.0
jsonschema-specifications 2024.10.1
locket                    1.0.0
lz4                       4.4.3
MarkupSafe                3.0.2
msgpack                   1.1.0
nchash                    0.1.5
netCDF4                   1.7.2
networkx                  3.4.2
numcodecs                 0.15.1
numpy                     2.2.2
packaging                 24.2
pandas                    2.2.3
partd                     1.4.2
pillow                    11.1.0
pip                       24.0
platformdirs              4.3.6
pluggy                    1.5.0
pooch                     1.8.2
psutil                    6.1.1
pyarrow                   19.0.0
pydantic                  2.10.6
pydantic_core             2.27.2
pytest                    8.3.4
python-dateutil           2.9.0.post0
pytz                      2025.1
PyYAML                    6.0.2
referencing               0.36.2
requests                  2.32.3
rpds-py                   0.22.3
setuptools                65.5.0
six                       1.17.0
sortedcontainers          2.4.0
tblib                     3.0.0
toolz                     1.0.0
tornado                   6.4.2
typing_extensions         4.12.2
tzdata                    2025.1
urllib3                   2.3.0
wrapt                     1.17.2
xarray                    2025.1.2
xyzservices               2025.1.0
yamanifest                0.3.8
zarr                      3.0.2
zict                      3.0.0
zipp                      3.21.0
``` 
TLDR here; `$ source venv/bin/activate && pip install pytest ~/access-nri-intake-catalog ~/intake-dataframe-catalog`

Annoyingly, it's necessary to run `python -m pytest` to get the right python interpreter running pytest. That's also the motivation behind the `python -c "import intake_dataframe_catalog; print(intake_dataframe_catalog.__file__)"`- PBS doesn't seem to reliably pick up the python version I'm after - maybe this is just a skill issue for me.

After all that, we have:
```bash
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/e2e/test_end_to_end.py::test_cmip5_metacat_vals_found[institute-expected2]
FAILED tests/e2e/test_end_to_end.py::test_cmip5_metacat_vals_found[model-expected3]
FAILED tests/e2e/test_end_to_end.py::test_cmip5_metacat_vals_found[experiment-expected5]
FAILED tests/e2e/test_end_to_end.py::test_cmip5_metacat_vals_found[table-expected7]
FAILED tests/e2e/test_end_to_end.py::test_cmip5_metacat_vals_found[ensemble-expected8]
FAILED tests/e2e/test_end_to_end.py::test_cmip5_metacat_vals_found[version-expected9]
FAILED tests/e2e/test_end_to_end.py::test_cmip5_metacat_vals_found[variable-expected10]
FAILED tests/e2e/test_end_to_end.py::test_cmip5_metacat_vals_found[time_range-expected11]
====== 8 failed, 100 passed, 1 xpassed, 72 warnings in 459.82s (0:07:39) =======
```

Why might these be failing, I hear you ask? When we just updated those tests on the 10th of February, to reflect the changes on the 9th?
```bash
$ cd /g/data/al33/catalog/v2/esm/
[ct1163@gadi-login-04 esm]$ ll
total 97616
-rw-r-----+ 1 jxg900 al33     1706 Feb 11 02:37 catalog.json
-rw-r-----+ 1 jxg900 al33 99947129 Feb 11 02:37 cmip5-al33.csv.gz
```
🤷‍♂️ 

Either way - the meaningful/expected to be correct e2e tests pass.

marc-white commented:
OK, glad that's now sorted. Two questions:

- Does the updated e2e test script need to be checking in?
- Are you happy including all the main updates in this release, or do you still want to revert and cherry-pick only the necessities?

charles-turner-1 commented:
> OK, glad that's now sorted. Two questions:
> 
> * Does the updated e2e test script need to be checking in?

Once everything is in main, that script should work fine. It's more an issue with the e2e test script being pinned to:
a. The branch checkout out in the `/g/data/xp65/admin/access-nri-intake-catalog` copy of the repo.
b. The dependencies contained within the conda/analysis3 environment that script uses: `module load conda/analysis3-24.12`

So I think, all in all, that we just leave that script as is until we have a better solution.... Messy I  know, but I can't think of a better way of handling it right now.

> * Are you happy including all the main updates in this release, or do you still want to revert and cherry-pick only the necessities?

I think we might as well just release this all as 1.1.0 to be honest - given how messy the process with the datastores & rebuilding the catalog seems to get, I think it's going to make more sense to avoid doing it twice.

charles-turner-1 commented:
I really should have double checked this was in... looks like we'll skip straight b0 straight to b1

charles-turner-1 commented:
@marc-white you able to do a quick review? Should be fairly trivial

marc-white reviewed (approved):
Yep, all looks fine.
